apps/system/js/icc_events.js

```diff
@@ -175,28 +174,21 @@ var icc_events = {
       ' - Events list:', eventList);
 
     var conn = icc.getConnection(message.iccId);
-    this.register_icc_event_voicechange = function(evt) {
-      icc_events.handleLocationStatus(message, evt);
-    };
     conn.removeEventListener('voicechange',
       this.register_icc_event_voicechange);
 
-    this.register_icc_event_idlescreen = function() {
-      icc_events.handleIdleScreenAvailableEvent(message);
-    };
+    if (this.stkUserActivity) {
+      navigator.removeIdleObserver(this.stkUserActivity);
+    }
```

Comment:
Here we'll remove the idle observer only when a new user activity handler is requested.
AFAIK, we should remove it after inform the SIM card of the first user activity. Correct me if I'm wrong